### PR TITLE
Reorder ch2

### PR DESCRIPTION
**Pull request by Nowosad**
@Robinlovelace please take a look, let me know what you think about it, and improve if needed. The main changes are:

1. I added a diagram showing the relationship between different sf building blocks (inspired by @defuneste)
2. I changed the order of sections -- now the sf section is before sfg and sfc sections. My thinking is that most people should know about sf, but not as many need to know about sfg...

image

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Taking a look :eyes:

File: 02-spatial-data.Rmd [hunk 1]
```diff
@@ -444,6 +444,74 @@ plot(st_as_sfc("GEOMETRYCOLLECTION (MULTIPOINT (5 2, 1 3, 3 4, 3 2), LINESTRING
...
 ```
 
+### The sf class {#sf}
+
+Simple features consist of two main parts: geometries and non-geographic attributes.
+Figure \@ref(fig:02-sfdiagram) shows how an sf object is created -- geometries come from an `sfc` object, while attributes are taken from a `data.frame` or `tibble`.
+To learn more about building sf geometries from scratch read the following Sections \@ref(sfg) and \@ref(sfc).
+
+```{r 02-sfdiagram, fig.cap="Building blocks on sf objects."}
+knitr::include_graphics("figures/02-sfdiagram.png")
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd:454]: Only comment here: shouldn't we add 

```r
# source("code/02-sfdiagram.R")
```

Above the line to show the code that generates the diagram?

File: 02-spatial-data.Rmd [hunk 2]
```diff
@@ -444,6 +444,74 @@ plot(st_as_sfc("GEOMETRYCOLLECTION (MULTIPOINT (5 2, 1 3, 3 4, 3 2), LINESTRING
 par(old_par)
 ```
 
+### The sf class {#sf}
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd:447]: Big :+1: to mentioning `sf` class first.

**Review by Robinlovelace — APPROVED** [review]
Approved with minor comment that could be tweaked in post merge commit (or not at all as it's such a minor comment). I think this is a clear win for readers.

**Comment by Nowosad** [comment]
Thanks, Robin.

**Pull request merged**